src/loggers/database-logger.php

```diff
@@ -0,0 +1,135 @@
+<?php
+/**
+ * A simple logger to allow query and memory usage logging.
+ *
+ * @package Yoast\WP\Free\Initializers
+ */
+
+namespace Yoast\WP\Free\Loggers;
+
+use Yoast\WP\Free\Conditionals\Development_Conditional;
+use Yoast\WP\Free\Integrations\Integration_Interface;
+use YoastSEO_Vendor\ORM;
+
+/**
+ * Class Database_Logger
+ */
+class Database_Logger implements Integration_Interface {
+	/**
+	 * @inheritDoc
+	 */
+	public function register_hooks() {
+		ORM::configure( 'logging', true );
+		ORM::configure( 'logger', [ $this, 'logger' ] );
+
+		add_action( 'shutdown', [ $this, 'log_output' ] );
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public static function get_conditionals() {
+		return [ Development_Conditional::class ];
+	}
+
+	/**
+	 * Private array of queries used for logging.
+	 *
+	 * @var array
+	 */
+	protected $query_log = [];
+
+	/**
+	 * Logs the query to a local variable for output on shutdown.
+	 *
+	 * @param string $query The query.
+	 * @param float  $time  The time the query took to execute.
+	 */
+	public function logger( $query, $time ) {
+		$query             = [
+			'query' => $query,
+			'time'  => $time
+		];
+		$this->query_log[] = $query;
+	}
+
+	/**
+	 * Outputs some logging.
+	 */
+	public function log_output() {
+		if (
+			( defined( 'DOING_AJAX' ) && DOING_AJAX ) ||
+			( defined( 'WP_CLI' ) && WP_CLI ) ||
+			( defined( 'REST_REQUEST' ) && REST_REQUEST )
+		) {
+			return;
+		}
+
+		echo PHP_EOL, PHP_EOL, '<!--';
+
+		$this->log_memory_usage();
+		$this->log_idiorm_queries();
+		$this->log_wpdb_queries();
+
+		echo '-->', PHP_EOL;
+	}
+
+	/**
+	 * Outputs a log header.
+	 *
+	 * @param $string
+	 */
+	private function header( $string ) {
+		echo PHP_EOL, PHP_EOL, $string, PHP_EOL;
+		echo '====', PHP_EOL;
+	}
+
+	/**
+	 * Logs the memory usage.
+	 *
+	 * @return void
+	 */
+	protected function log_memory_usage() {
+		$memory_used_peak = number_format( ( memory_get_peak_usage() / 1048576 ), 2 );
```

Comment:
To make it directly readable a descriptive named variable containing the value `1048576` would be nice. For me personally it's difficult to process.